### PR TITLE
[RDY] Discussion & Recommendations

### PR DESCRIPTION
**Pull request by Hjdskes**
I have slightly restructured the recommendations section, too.

Branched from #66.

### CODE REVIEW

File: final/discuss-recommend/eclipse.tex [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+The delivered Eclipse frontend is just that: a frontend to the REPL backend.
+Integration with Eclipse and Spoofax Eclipse are not yet present. This
+subsection provides several recommendations to better integrate this frontend
+into Eclipse and Spoofax Eclipse.
+
+\subsubsection{Building languages and projects}
+
+TODO: write this when the discussion on classpath issues has been written.
+Mention that there should be a button to build the language/project, put it on
+the classpath and launch it in a REPL.
```
Comment (Balletie) [inline final/discuss-recommend/eclipse.tex]: Or suggest OSGi bundles as a possible solution

Comment (Hjdskes) [inline final/discuss-recommend/eclipse.tex]: I can do this, but this won't be a general solution to the problem: the console frontend and future other frontends likely aren't using OSGi.

File: final/discuss-recommend.tex [hunk 2]
```diff
@@ -1,6 +1,60 @@
...
+TODO: introduction
+
+\input{discuss-recommend/repl}
+
+\input{discuss-recommend/alternate-eval}
+
+\subsection{Improvements to the Eclipse frontend}
+\label{ssec:impr-eclipse}
+
+\input{discuss-recommend/eclise}
```
Comment (gfokkema) [inline final/discuss-recommend.tex]: s/eclise/eclipse

File: final/discuss-recommend.tex [hunk 3]
```diff
@@ -12,15 +12,44 @@ \chapter{Discussion and Recommendations}
...
+not been implemented (see \cref{ssec:eval-requirements}) and opportunities for
+improvements in the delivered product. First, recommendations are made to
+improve the REPL backend itself. Secondly, suggestions are made on
+how to improve the Eclipse frontend. Finally, recommendations to provide
+additional frontend implementations are made.
+
+\subsection{Extending the functionality of the REPL}
+\label{ssec:impr-backend}
+
+The delivered backend is a solid product. Due to time constrains, however, some
```
Comment (Hjdskes) [inline final/discuss-recommend.tex]: contraints

File: final/discuss-recommend.tex [hunk 4]
```diff
@@ -12,15 +12,44 @@ \chapter{Discussion and Recommendations}
...
+nice to have features have not been implemented. This subsection shortly
```
Comment (Hjdskes) [inline final/discuss-recommend.tex]: Find a synonym for "nice to have"

File: final/discuss-recommend.tex (hunk 3 above)
Comment (gfokkema) [inline final/discuss-recommend.tex]: *constraints lol

File: final/discuss-recommend.tex (hunk 4 above)
Comment (gfokkema) [inline final/discuss-recommend.tex]: desirable

Comment (Hjdskes) [inline final/discuss-recommend.tex]: That's even better than optional, I'll update that :)

File: final/recommendations/eclipse.tex [hunk 5]
```diff
@@ -28,10 +37,12 @@ \subsection{Improving the UI of the Eclipse REPL}
...
 typical REPL: the input and output views are separated. To improve the user
-experience, these two views could be merged into one. TODO: expand and give
-pointers as to how this might be possible?
+experience, these two views could be merged into one. This would require the use
+of a widget or text representation that can be \textit{``semi-read-only''}, as
+the previously entered text and results should not be editable, while the
+current input should be.
 
-TODO: recommend missed features such as autocompletion et cetera?
+TODO: recommend missed features such as autocompletion, hovers, et cetera?
```
Comment (Hjdskes) [inline final/recommendations/eclipse.tex]: What do you think?

File: final/recommendations.tex [hunk 6]
```diff
@@ -0,0 +1,33 @@
+\chapter{Recommendations}
+\label{cha:recommendations}
+
+This chapter makes several recommendations to the client for future work on the
+delivered product. First, recommendations are made to improve the REPL backend
+itself. Secondly, suggestions are made on how to improve the Eclipse frontend.
+Finally, a recommendation to provide literal programming is made.
```
Comment (gfokkema) [inline final/recommendations.tex]: > *on how to provide literate programming functionality?

also, s/literal/literate

File: final/recommendations/eclipse.tex [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+\subsection{Improving the UI of the Eclipse REPL}
+
+The user interface currenty offered by the Eclipse frontend does not resemble a
+typical REPL: the input and output views are separated. To improve the user
+experience, these two views could be merged into one. This would require the use
+of a widget or text representation that can be \textit{``semi-read-only''}, as
+the previously entered text and results should not be editable, while the
+current input should be.
+
+TODO: recommend missed features such as autocompletion, hovers, et cetera?
```
Comment (gfokkema) [inline final/recommendations/eclipse.tex]: I'd just leave this TODO out tbh. You've already shortly mentioned hovers somewhere else.

File: final/discussion/eclipse.tex [hunk 8]
```diff
@@ -0,0 +1,77 @@
+\section{Eclipse multithreading issues}
+\label{sec:eclipse-multithread}
+
+The implementation of the Eclipse frontend has been a source of exposing many
+shortcomings in the initial designs. These shortcomings and how they have been
+resolved are discussed in this subsection.
```
Comment (Balletie) [inline final/discussion/eclipse.tex]: section

**Pull request merged**